src/Func.h

```diff
@@ -833,7 +833,7 @@ class Func {
                               const std::vector<ExternFuncArgument> &params,
                               Type t,
                               int dimensionality) {
-        define_extern(function_name, params, Internal::vec<Type>(t), dimensionality);
+        define_extern(function_name, params, std::vector<Type>{t}, dimensionality);
```

Comment:
I gather the explicit cast to vector here is needed because define_extern has an overload for both Type and std::vector<Type>. Given the convenient syntax, do we want to get rid of the overload? Or perhaps we should introduce introduce a proxy class so all of define_extern(..., t, ...), define_extern(..., {t}, ...), and define_extern(..., {t1, t2, t3}, ...) work?